sites/main-site/src/content/blog/2024/seqera-containers-part-1.mdx

```diff
@@ -0,0 +1,453 @@
+[generate the name of the mulled container](https://midnighter.github.io/mulled) which gently guides you through
+the process of finding the name of your container and how to update a container if you want to bump the software versions.
+
+Once the image is built and you know its address, the final step is to go to nf-core/modules,
+create a pull-request with the updated containers, and bump the versions in the conda `environment.yml`.
+:::
+
+This system works and, despite its complexity, is now a familiar process to many nf-core developers.
+However, it does present some problems.
+It's a highly manual process to update software versions and the container declarations - just fetching the software for a module often ends up taking longer than writing the entire module.
+There are also significant delays in waiting for the images to become available.
+All in all, it can be a frustrating experience and we see this in the volume of Slack messages asking for help.
+It seems likely that this puts some people off from contributing to nf-core.
+
+Migrating nf-core to use Seqera Containers will make provisioning multi-package containers trivial.
+The module developer will only need to edit the module `environment.yml` file and everything else will be fully automated and made available almost immediately.
+Packages can be used minutes after their release on Bioconda and we will have automation in place to bump the Conda + Seqera Containers packages when a tool is updated.
+
+### Time to Singularity image
+
+BioContainers builds both Docker and Singularity images.
+Docker images are pushed to quay.io and are available almost immediately,
+whereas Singularity images are pushed to the Galaxy FTP server in a nightly build, taking up to 24 hours to become available.
+
+This delay is frustrating for developers, as it breaks the flow of working on an update.
+Once a new version of a tool is released, they have to wait for the Singularity image to become available before they can add it and test the module.
+In practice, this often means that module updates get stuck in limbo for days until the developer has time to come back and check if the image is available.
+
+Singularity images are generated on the fly with Seqera Containers.
+Developers will be able update the nf-core module minutes after the Bioconda package is released.
+
+### BioContainers API
+
+To simplify adding single-package BioContainer images to nf-core modules, we have developed the nf-core/tools package to query the BioContainer API for a given tool version and fetch the image URIs.
+This process is repeated during linting of nf-core modules and pipelines, to ensure that there is not an accidental mismatch between Conda and Docker / Singularity software versions.
+Unfortunately, the BioContainer API has had a history of being slow and unreliable, with a lot of down time (204 error message anyone?).
```

Comment:
ACtually it might be the quay.io side of things resulting in the 204...

Reply:
Ok, that was from @edmundmiller's original text. Will remove if we're not sure (also now feels a little different tone to the rest anyway).

```suggestion
Unfortunately, the BioContainer API has had a history of being slow and unreliable, with a lot of down time.
```